Testscripts/Linux/ETHTOOL-OFFLOADING-SETTING.sh

```diff
@@ -0,0 +1,261 @@
+    LogMsg  "Synthetic interface found: $synthetic_interface_vm_1"
+    if [[ $DISTRO_VERSION =~ ^6\. ]]; then
+        synthetic_MAC=$(ip link show ${synthetic_interface_vm_1} | grep ether | awk '{print $2}')
+        vf_interface_vm_1=$(grep -il ${synthetic_MAC} /sys/class/net/*/address | grep -v $synthetic_interface_vm_1 | sed 's/\// /g' | awk '{print $4}')
+    else
+        vf_interface_vm_1=$(find /sys/devices/* -name "*${synthetic_interface_vm_1}" | grep "pci" | sed 's/\// /g' | awk '{print $12}')
+    fi
+    LogMsg "Virtual function found: $vf_interface_vm_1"
+
+    # Ping the remote host
+    i=0
+    ping_exit_code=1
+    while [ $i -lt 4 ]
+    do
+        i=$(($i+1))
+        sleep 10
+        ping -c 11 "$static_IP_2" > "$static_IP_2.log" 2>&1
+        ping_exit_code=$?
+        LogMsg "Attempt #$i to ping $VF_IP2 through $synthetic_interface_vm_1"
+        if [ "$ping_exit_code" == 0 ]; then
+            LogMsg "Successfully pinged $VF_IP2 through $synthetic_interface_vm_1"
+            break
+        fi
+    done
+
+    if [ "$ping_exit_code" == 1 ]; then
+        LogErr "Unable to ping $VF_IP2 through $synthetic_interface_vm_1"
+        SetTestStateFailed
+        exit 0
+    fi
+
+    # Take a note on the offloading state and set it on
+    for key in ${!offload_features[*]}
+    do
+        on_or_off=$(ethtool -k $synthetic_interface_vm_1 2>&1 | grep "^$key" | awk {'print $2'})
+        fixed_or_not=$(ethtool -k $synthetic_interface_vm_1 2>&1 | grep "^$key" | awk {'print $3'})
+        LogMsg "The initial status of $key for $synthetic_interface_vm_1: $on_or_off $fixed_or_not"
+        # Set the offload on if its status is off and not fixed
+        if [[ "$on_or_off" == "off" && x"$fixed_or_not" != x"[fixed]" ]];then
+            LogMsg "Set ${offload_features[$key]} on"
+            set_feature_status $synthetic_interface_vm_1 ${offload_features[$key]} "on"
+        fi
+    done
+
+    # The below is against scatter-gather feature
+    # Step 1: Check the status of scatter-gather
+    feature_name="scatter-gather"
+    feature_device_name="sg"
+    check_feature_status $synthetic_interface_vm_1 $feature_name "on"
+
+    # Step 2: Check sync offloading features to VF NIC
+    check_feature_status $vf_interface_vm_1 $feature_name "on"
+
+    # Step 3: Check scatter-gather feature to be tunable
+    set_feature_status $synthetic_interface_vm_1 $feature_device_name "off"
+    # Check sync offloading features to VF NIC again
+    check_feature_status $vf_interface_vm_1 $feature_name "off"
+
+    # Step 4: Disable/enable SRIOV
+    if ! DisableEnablePCI "SR-IOV"; then
+        LogErr "Could not disable and reenable pci device."
+        SetTestStateFailed
+        exit 0
+    fi
+
+    # Step 5: Verify the changed setting persists on VF
+    check_feature_status $synthetic_interface_vm_1 $feature_name "off"
+    check_feature_status $vf_interface_vm_1 $feature_name "off"
```

Comment:
Should we check other feature_names, which are not scatter-gather,  the expected is they are using 'default' values.  only scatter-gather is tunable, and always synchronized from synthetic NIC to the VF NIC during ndo_set_features (ethtool -K), is that right?

Reply:
I think 'sg' is mandatory based on @haiyangz feedback. Others should be optional.

Reply:
My point is we should design our case pass only be restricted to a non-default settings detected after DisableEnablePCI.   e.g.,  DisableEnablePCI  for all, check if all feature names have default value ('on' probably), then  set scatter-gather with non-default values through synthetic only,  DisableEnablePCI  again for all,  check VF of scatter-gather feature name which already synced from synthetic settings with the non-default value.